packages/server-wallet/src/wallet/index.ts

```diff
@@ -391,6 +391,44 @@ export class Wallet extends EventEmitter<WalletEvent>
     return this.store.lockApp(channelId, criticalCode, handleMissingChannel);
   }
 
+  async closeChannels(channelIds: Bytes32[]): Promise<MultipleChannelOutput> {
+    for (const channelId of channelIds) {
+      await this.store.lockApp(
+        channelId,
+        async (tx, channel) => {
+          // TODO: (Objectives Rewrite) Keeping this here b/c we need to do input validation
+          // and check if its our turn and throw an error as existing tests expect
+          getOrThrow(CloseChannel.closeChannel(channel));
+
+          await this.store.addObjective(
+            {
+              type: 'CloseChannel',
+              participants: [],
+              data: {targetChannelId: channelId, fundingStrategy: channel.fundingStrategy},
+            },
+            tx
+          );
+        },
+        () => {
+          throw new CloseChannel.CloseChannelError(
+            CloseChannel.CloseChannelError.reasons.channelMissing,
+            {channelId}
+          );
+        }
+      );
+    }
+
+    const {channelResults, outbox} = await this.takeActions(channelIds);
+
+    (outbox[0].params.data as Payload).objectives = channelIds.map(channelId => ({
```

Comment:
Is it possible the type assertion?

Reply:
The `Store` creates an outbound objective right now and the type of this is `unknown` which is strategically set (cc @tomclose) for outbound objects. One solution might be to put this line of code inside the store... but that would create an even bigger mess and contribute to the code smell of the store handling communications. I think this type assertion is OK for now.

Reply:
Seems to me that the wallet _should_ be using the `Payload` type. It should be `unknown` only from the point of view of the app / messaging layer (outside the wallet's interface).

Reply:
There should probably be a cleanup of the wallet code and `takeActions` to account for this, maybe a new issue